authorization/v2/authorizer_test.go

```diff
@@ -0,0 +1,236 @@
+				assert.Equal(t, testUserId.Hex(), claims.Id)
+			},
+		},
+		{
+			name: "should use correct IssuedAt",
+			checks: func(claims TokenClaims) {
+				assert.Equal(t, testTimestamp.Unix(), claims.IssuedAt)
+			},
+		},
+		{
+			name: "should use correct ExpiresAt",
+			checks: func(claims TokenClaims) {
+				assert.Equal(t, testTimestamp.Unix()+testTTL, claims.ExpiresAt)
+			},
+		},
+		{
+			name: "should use correct TokenType",
+			checks: func(claims TokenClaims) {
+				assert.Equal(t, user, claims.TokenType)
+			},
+		},
+	}
+
+	jwtSecret := "test_secret"
+	authorizer, timeProvider := setupAuthorizerTests(t, jwtSecret)
+	timeProvider.EXPECT().Now().Return(testTimestamp).Times(1)
+
+	token, err := authorizer.CreateUserToken(testUserId, testTimestamp.Unix()+testTTL)
+	assert.NoError(t, err)
+
+	claims := extractTokenClaims(t, token, jwtSecret)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.checks(claims)
+		})
+	}
+}
+
+func TestAuthorizer_GetAuthorizedResources_should_return_correct_uris_when_token_is_valid(t *testing.T) {
+	jwtSecret := "test_secret"
+	authorizer, _ := setupAuthorizerTests(t, jwtSecret)
+	token := createToken(t, "testuser", nil, int64(100), user, jwtSecret)
+	uris := []UniformResourceIdentifier{"test"}
+
+	returnedUris, err := authorizer.GetAuthorizedResources(token, uris)
+	assert.NoError(t, err)
+
+	assert.Equal(t, uris, returnedUris)
+}
+
+func TestAuthorizer_GetAuthorizedResources_should_return_err(t *testing.T) {
+	jwtSecret := "jwtSecret"
+
+	tests := []struct {
+		name      string
+		token     string
+		wantedErr error
+	}{
+		{
+			name:      "when token is invalid",
+			token:     "invalid token",
+			wantedErr: ErrInvalidToken,
+		},
+		{
+			name:      "when token type is invalid",
+			token:     createToken(t, "user id", nil, int64(0), "unknown type", jwtSecret),
+			wantedErr: ErrInvalidToken,
+		},
+		{
+			name:      "when token is expired",
+			token:     createToken(t, "user id", nil, int64(-5), user, jwtSecret),
+			wantedErr: ErrInvalidToken,
+		},
+	}
+
+	authorizer, _ := setupAuthorizerTests(t, jwtSecret)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			uris, err := authorizer.GetAuthorizedResources(tt.token, nil)
+			assert.Nil(t, uris)
+			assert.Equal(t, tt.wantedErr, errors.Cause(err))
+		})
+	}
+}
+
+func Test_verifyTokenType(t *testing.T) {
+	tests := []struct {
+		tokenType TokenType
+		wantErr   bool
+	}{
+		{
+			tokenType: user,
+		},
+		{
+			tokenType: service,
+		},
+		{
+			tokenType: "unknown type",
+			wantErr:   true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(string(tt.tokenType), func(t *testing.T) {
+			assert.Equal(t, tt.wantErr, verifyTokenType(tt.tokenType) != nil)
+		})
+	}
+}
+
+func createToken(t *testing.T, id string, allowedResources []UniformResourceIdentifier, timeToLive int64, tokenType TokenType, jwtSecret string) string {
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, TokenClaims{
```

Comment:
As above, this could then be replaced to use the same signing method as in `authorization/v2/authorizer.go `